src/java.desktop/unix/classes/sun/awt/screencast/ScreencastHelper.java

```diff
@@ -105,11 +114,30 @@ private static List<Rectangle> getSystemScreensBounds() {
                 ).toList();
     }
 
+    private static synchronized native void closeSession();
+
+    private static void timerCloseSessionRestart() {
+        if (timerTask != null) {
+            timerTask.cancel();
+        }
+
+        timerTask = new TimerTask() {
+            @Override
+            public void run() {
+                closeSession();
```

Comment:
What happen if the "closeSession" is executed why the new Robot#getRGBPixels is called? In the robot code we call timerCloseSessionRestart to restart/cancel timer, but I think it could be possible that closeSession() is already executed and may close session right before it will be used by the robot but after we "open" it. 

Or this case is not possible?

Reply:
Both `ScreencastHelper#getRGBPixels`(`Robot#getRGBPixels` calls it internally) and `ScreencastHelper#closeSession` are synchronized. 
So `ScreencastHelper#closeSession` will wait until the execution of `ScreencastHelper#getRGBPixels` is finished and vice versa.

Many hours of testing with random delays close to DELAY_BEFORE_SESSION_CLOSE worked fine.

Reply:
Java part looks good to me. Why 2s and not more or less ? 
The delay could be set through a system property ?

Reply:
>  Why 2s and not more or less ?

This is the average value chosen so as not to slow down test execution, and not to annoy the user with the "Screen is being shared" icon ![image](https://github.com/openjdk/jdk/assets/77687766/ded56ab9-c38d-4640-af8b-e2e116f6779e)

> The delay could be set through a system property ?

This functionality could be added, but I don't see much reason to do so. 
If a screenshot is taken occasionally, the benefit is quite a bit compared to frequent requests.
Plus it requires some CSR work.